trpc/queries/goals.ts

```diff
@@ -598,8 +563,22 @@ export const addCalculatedGoalsFields = (goal: any, activityId: string, role: Ro
     const _shortId = `${goal.projectId}-${goal.scopeId}`;
     const _hasAchievementCriteria = !!goal.goalAchiveCriteria?.length;
     const _achivedCriteriaWeight: number | null =
-        goal.completedCriteriaWeight == null && goal.goalAchieveCriteria
-            ? calcAchievedWeight(goal.goalAchieveCriteria)
+        goal.completedCriteriaWeight == null && goal.goalAchieveCriteria?.length
+            ? calcAchievedWeight(
+                  goal.goalAchieveCriteria.map(
+                      ({
+                          weight,
+                          isDone,
+                          deleted,
+                          criteriaGoal,
+                      }: GoalAchieveCriteria & { criteriaGoal: (Goal & { state: State | null }) | null }) => ({
```

Comment:
Can we use the same structure in calcAchievedWeight and avoid map for every usage?